src/components/UpstreamDetector/UpstreamDetector.tsx

```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+
+interface UpstreamDetectorProps {
+  readonly upstream: React.Component;
+  readonly downstream: React.Component;
+}
+
+export const isUpstream = process.env.REACT_APP_IS_UPSTREAM === 'true';
+
+export default class UpstreamDetector extends React.PureComponent<UpstreamDetectorProps> {
```

Comment:
It's not being used at the moment, right?

Reply:
Right, in the end I didn't use this. Could be useful for https://issues.jboss.org/browse/KIALI-3131 or we could remove it.